compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -507,4 +491,254 @@ LogicalResult splitLogicalObjectFifos(
+             << splitDimTarget;
+    }
+    if (targetSize.value() != 1) {
+      return producer.emitOpError() << "only a static size of 1 is currently "
+                                       "supported on the split index";
+    }
+    assert(targetOffset < newObjFifos.size() &&
+           "the targetOffset should be smaller than the number of objectFifos");
+
+    // The index has been encoded as target offset for the split dimension, so
+    // fetch the corresponding objectFifo based on it.
+    AMDAIE::LogicalObjectFifoFromMemrefOp newObjFifo =
+        newObjFifos[targetOffset.value()];
+    targetOffsets[splitDimTarget] = rewriter.getIndexAttr(0);
+    rewriter.setInsertionPoint(producer);
+    auto newDmaOp = rewriter.create<AMDAIE::DmaCpyNdOp>(
+        producer.getLoc(), newObjFifo, targetOffsets, targetSizes,
+        targetStrides, producer.getSource(), producer.getSourceMixedOffsets(),
+        producer.getSourceMixedSizes(), producer.getSourceMixedStrides());
+    rewriter.replaceOp(producer, newDmaOp);
+  }
+
+  // Update the consumer dma ops.
+  for (AMDAIE::DmaCpyNdOp consumer : consumers) {
+    SmallVector<OpFoldResult> sourceOffsets = consumer.getSourceMixedOffsets();
+    SmallVector<OpFoldResult> sourceSizes = consumer.getSourceMixedSizes();
+    SmallVector<OpFoldResult> sourceStrides = consumer.getSourceMixedStrides();
+
+    std::optional<int64_t> sourceSize =
+        getConstantIntValue(sourceSizes[splitDim]);
+    std::optional<int64_t> sourceOffset =
+        getConstantIntValue(sourceOffsets[splitDim]);
+    if (!sourceSize || !sourceOffset) {
+      return consumer.emitOpError()
+             << "expected a static source offset and size on index: "
+             << splitDim;
+    }
+    if (sourceSize.value() != 1) {
+      return consumer.emitOpError() << "only a static size of 1 is currently "
+                                       "supported on the split index";
+    }
+    assert(sourceOffset < newObjFifos.size() &&
+           "the sourceOffset should be smaller than the number of objectFifos");
+
+    // The index has been encoded as source offset for the split dimension, so
+    // fetch the corresponding objectFifo based on it.
+    AMDAIE::LogicalObjectFifoFromMemrefOp newObjFifo =
+        newObjFifos[sourceOffset.value()];
+    sourceOffsets[splitDim] = rewriter.getIndexAttr(0);
+    rewriter.setInsertionPoint(consumer);
+    auto newDmaOp = rewriter.create<AMDAIE::DmaCpyNdOp>(
+        consumer.getLoc(), consumer.getTarget(),
+        consumer.getTargetMixedOffsets(), consumer.getTargetMixedSizes(),
+        consumer.getTargetMixedStrides(), newObjFifo, sourceOffsets,
+        sourceSizes, sourceStrides);
+    rewriter.replaceOp(consumer, newDmaOp);
+  }
+
+  return success();
+}
+
+/// Split DmaCpyNd ops between L2 and L3 memory spaces.
+LogicalResult splitDoublyStridedOp(IRRewriter &rewriter,
+                                   AMDAIE::DmaCpyNdOp op) {
```

Comment:
This function will be easier to reuse and understand if the signature is 

```
LogicalResult splitDoublyStridedOp(IRRewriter &rewriter, AMDAIE::DmaCpyNdOp op, uint32_t splitDimension, uint32_t splitSize, bool splitIsOnTargetSize); 
```

Please make this change.

Reply:
I prefer not doing this. splitDim, splitFactor and splitDimTarget are determined by the input DmaCpyNdOp, each DmaCpyNdOp could have different sets of those values. I could split this function into two parts that first acquiring these values from the input dma and then pass the parameters with this dma to `splitDoublyStridedOp` again, but I think it would make the function call in `AMDAIESplitLogicalObjFifos` more complicated.

Reply:
Yeah, I also feel like this function needs to be split up into one function that decides which DMA copy ops to split (and what the splitting factor should be) and another one that does the actual splitting.

Reply:
Okay, done.